### ISSUE TITLE
How to make own grammar ?

### ISSUE THREAD

**Issue by michalss** [AvaloniaUI/AvaloniaEdit#385]
Hi is there any docs or sample how to make own grammer please ? I would need modify c# grammer and made some changes. How to do it pls ? There is not any docs.. :(

Regards

**Comment by danipen** [comment]
Grammars are provided by [TextmateSharp](https://github.com/danipen/TextMateSharp) repository.

**Comment by danipen** [comment]
Yes so you'd need to base on the [existing TextmateSharp c# grammar](https://github.com/danipen/TextMateSharp/tree/master/src/TextMateSharp.Grammars/Resources/Grammars/csharp) and make your changes from there. No docs are provided 😔

**Comment by danipen** [comment]
Yes you need to build custom `IRegistryOptions` see `LocaRegistryOptions` usage [here](https://github.com/danipen/TextMateSharp/tree/master#using).

**Comment by Super1Windcloud** [comment]
Does textmate only support C#? When I use other language extensions, nothing happens. It’s still C#’s syntax highlighting.
